windows/setup.ps1

```diff
@@ -17,14 +17,41 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
-# $params is not currently used in this module
-# $params = Parse-Args $args;
+# enabled $params (David O'Brien, 06/08/2015)
+$params = Parse-Args $args;
+
+
+Function Get-CustomFacts {
+  [cmdletBinding()]
+  param (
+    [Parameter(mandatory=$false)]
+    $factpath = $null
+  )
+
+  if (-not (Test-Path -Path $factpath)) {
+    Fail-Json $result "The path $factpath does not exist. Typo?"
+  }
+
+  $FactsFiles = Get-ChildItem -Path $factpath | Where-Object -FilterScript {($PSItem.PSIsContainer -eq $false) -and ($PSItem.Extension -eq '.ps1')}
+
+  foreach ($FactsFile in $FactsFiles) {
+      $out = . $($FactsFile.FullName)
```

Comment:
Do you have a specific reason for dot-sourcing instead of call-ing (&)? Seems like the definition persistence behavior of dot-sourcing would be a negative here when we're likely only interested in the result (potential to collide with built-in funcs, etc).